docs/data/charts/line-demo/LiveLineChartNoSnap.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { LineChart } from '@mui/x-charts/LineChart';
+
+const xLabels = [
+  'Page A',
+  'Page B',
+  'Page C',
+  'Page D',
+  'Page E',
+  'Page F',
+  'Page G',
+];
+
+const randBetween = (min: number, max: number) => {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+};
+
+export default function LiveLineChartNoSnap() {
+  const [uData, setUData] = React.useState([
+    4000, 3000, 2000, 2780, 1890, 2390, 3490,
+  ]);
+  const [pData, setPData] = React.useState([
+    2400, 1398, 9800, 3908, 4800, 3800, 4300,
+  ]);
+
+  React.useEffect(() => {
+    const intervalId = setInterval(() => {
+      setUData((prev) => [...prev.slice(1), randBetween(0, 5000)]);
+      setPData((prev) => [...prev.slice(1), randBetween(0, 5000)]);
+    }, 1000);
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, []);
+
+  return (
+    <LineChart
+      height={300}
+      series={[
+        { data: pData, label: 'pv' },
+        { data: uData, label: 'uv' },
+      ]}
+      xAxis={[{ scaleType: 'point', data: xLabels }]}
```

Comment:
To make it more natural, you could

- add `skipAnimation`
- Put more data points, such that one updates change less the overall shape. On this topic, a way to have more natural shape is to use random walks. The new element to the array is the last value + a random value `[...prev.slice(1),  prev[prev.length - 1] - 5 + 10 * Math.random()]`
- Update the x-axis too

And using dates in the x-axis that you update too
https://github.com/mui/mui-x/pull/17039

Reply:
Thanks! Incorporated some suggestions, however I think it makes sense to keep the animation, doesn't it? When data is more closely related, I think it looks good:


https://github.com/user-attachments/assets/25741bfd-ee69-4653-89ad-1aa417e09145

However, I think we should reconsider our line animation, as it didn't work very well when new data points are different from the last ones.

Reply:
We don't have a in/out behaviour built in, I suppose this would be a big topic by itself